desktop/cmp/tab/TabSwitcher.js

```diff
@@ -28,51 +34,249 @@ export const [TabSwitcher, tabSwitcher] = hoistCmp.withFactory({
     model: uses(TabContainerModel),
     className: 'xh-tab-switcher',
 
-    render({model, className, animate, ...props}) {
-        const {id, tabs, activeTabId} = model;
+    render({
+        model,
+        className,
+        orientation = 'top',
+        animate = false,
+        enableOverflow = false,
+        tabSize,
+        tabMinSize,
+        tabMaxSize
+    }) {
+        throwIf(!['top', 'bottom', 'left', 'right'].includes(orientation), 'Unsupported value for orientation.');
 
-        const orientation = withDefault(props.orientation, 'top'),
-            vertical = ['left', 'right'].includes(orientation);
+        const {id, tabs, activeTabId} = model,
+            vertical = ['left', 'right'].includes(orientation),
+            impl = useLocalModel(() => new LocalModel(model, enableOverflow, vertical));
+
+        // Implement overflow
+        let ref = impl.switcherRef;
+        if (impl.enableOverflow) {
+            ref = composeRefs(
+                impl.switcherRef,
+                useOnResize(() => impl.updateOverflowTabs()),
+                useOnVisibleChange(() => impl.updateOverflowTabs()),
+                useOnScroll(() => impl.updateOverflowTabs())
+            );
+        }
+
+        // Create tabs
+        const tabStyle = {};
+        if (isFinite(tabSize)) tabStyle[vertical ? 'height' : 'width'] = tabSize + 'px';
```

Comment:
should this always be width, even for vertical tabs? Seems like with the horizontal text we use for tabs, we always want to be manipulating the width.  will that work with the flexing in vertical orientations?

If we specify the new prop as xxxWidth, that also always leaves us the option for specifying the height in the future if we ever decide that makes sense.

Reply:
Very good point!

Although in vertical layout, setting the width of the tabs doesn't really make sense - they fill the width of the switcher container.